From 2b930f694c2275b892772857002724b9fdcae6c5 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Stefan=20Br=C3=BCns?= <stefan.bruens@rwth-aachen.de>
Date: Thu, 16 Jul 2020 03:52:23 +0200
Subject: [PATCH 1/2] Correct GL_BACK/GL_BACK_LEFT mapping on GLES

GLES does not have LEFT/RIGHT targets for the default framebuffer, but
just GL_BACK, so there is no need to remap it. Desktop GL may have
LEFT and RIGHT attached at the simultaneously, so one of the buffers
has to be selected when querying the attributes.
---
 Rendering/OpenGL2/vtkOpenGLRenderWindow.cxx | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/Rendering/OpenGL2/vtkOpenGLRenderWindow.cxx b/Rendering/OpenGL2/vtkOpenGLRenderWindow.cxx
index 9bbbc2ba54..4bda9330b6 100644
--- a/Rendering/OpenGL2/vtkOpenGLRenderWindow.cxx
+++ b/Rendering/OpenGL2/vtkOpenGLRenderWindow.cxx
@@ -719,6 +719,10 @@ int vtkOpenGLRenderWindow::GetColorBufferSizes(int* rgba)
 #ifdef GL_DRAW_BUFFER
     glGetIntegerv(GL_DRAW_BUFFER, &attachment);
 #endif
+#ifdef GL_ES_VERSION_3_0
+    // GLES only has the GL_BACK color
+    // attachment for the default framebuffer
+#else
     // GL seems odd with its handling of left/right.
     // if it says we are using GL_FRONT or GL_BACK
     // then convert those to GL_FRONT_LEFT and
@@ -731,6 +735,7 @@ int vtkOpenGLRenderWindow::GetColorBufferSizes(int* rgba)
     {
       attachment = GL_BACK_LEFT;
     }
+#endif
 
     // make sure we clear any errors before we start
     // otherwise we may get incorrect results
-- 
2.27.0

